source/module_elecstate/module_charge/charge_init.cpp

```diff
@@ -53,61 +53,37 @@ void Charge::init_rho(elecstate::efermi& eferm_iout,
             {
                 std::stringstream ssc;
                 ssc << PARAM.globalv.global_readin_dir << "SPIN" << is + 1 << "_CHG.cube";
-                double& ef_tmp = eferm_iout.get_ef(is);
-                if (ModuleIO::read_cube(
-#ifdef __MPI
-                    & (GlobalC::Pgrid),
-#endif
```

Comment:
If this change does not trigger any error, I guess no matter whether the macro __MPI is defined, the instance of class GlobalC::Pgrid is always created?

Reply:
Yes. Without MPI, It's just a lightweight struct of grid sizes.
I think `Grid` should be the better name of this class (instead of `Parallel_Grid`) in the future.

Reply:
@maki49 no...`Grid` is better to be a namespace rather than class, see: https://github.com/deepmodeling/abacus-develop/blob/fe3b44fe8c9b04195632ea844e32dc9cc19c5ae4/source/module_base/grid/radial.h#L6

Reply:
Got it. At least I will not rename this class in this PR.